src/extension.ts

```diff
@@ -176,8 +172,9 @@ async function startMachine(
   const startTime = performance.now();
 
   try {
-    await execMacadam(['start'], machineInfo.vmType, {
-      logger: new LoggerDelegator(context, logger),
+    await macadam.startVm({
```

Comment:
I think we should add a timeout for this as well as I'm getting issues of this being endlessly stuck, if it timeouts, we should output the stderr / stdout.

Reply:
@cdrage could you please have a look at this PR? I would prefer to merge it as is (the happy path is correctly working as far as I can see), and we can iterate to fix issues and cover new features in parallel